src/Expectation.php

```diff
@@ -52,6 +52,347 @@ public function toBe($value): Expectation
+     */
+    public function toContainOnlyInstancesOf(string $instance): Expectation
+    {
+        Assert::assertContainsOnlyInstancesOf($instance, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Assert that needles is a substring of value.
+     */
+    public function toContainString(string $needle): Expectation
+    {
+        Assert::assertStringContainsString($needle, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Assert that needles is a substring of value, ignoring the
+     * difference in casing.
+     */
+    public function toContainStringIgnoringCase(string $needle): Expectation
+    {
+        Assert::assertStringContainsStringIgnoringCase($needle, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Assert that $count matches the number of elements of $value.
+     */
+    public function toCount(int $count): Expectation
+    {
+        Assert::assertCount($count, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Asserts that two variables have the same value.
+     *
+     * @param mixed $value
+     */
+    public function toEqual($value): Expectation
+    {
+        Assert::assertEquals($value, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Asserts that two variables are equals, ignoring the casing
+     * for the comparison.
+     *
+     * @param mixed $value
+     */
+    public function toEqualIgnoringCase($value): Expectation
+    {
+        Assert::assertEqualsIgnoringCase($value, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Asserts that two variables have the same value.
+     * The contents of $expected and $actual are canonicalized before
+     * they are compared. For instance, when the two variables $value and
+     * $this->value are arrays, then these arrays are sorted before they are
+     * compared. When $value and $this->value are objects,
+     * each object is converted to an array containing all private,
+     * protected and public attributes.
+     *
+     * @param mixed $value
+     */
+    public function toEqualCanonicalizing($value): Expectation
+    {
+        Assert::assertEqualsCanonicalizing($value, $this->value);
+
+        return $this;
+    }
+
+    /**
+     * Assert that the absolute difference between $value and $this->value
+     * is greater than $delta.
+     *
+     * @param mixed $value
+     */
+    public function toEqualWithDelta($value, float $delta): Expectation
+    {
+        Assert::assertEqualsWithDelta($value, $this->value, $delta);
+
+        return $this;
+    }
+
+    /**
+     * Assert that the value infinite.
+     */
+    public function toBeInfinite(): Expectation
+    {
+        Assert::assertInfinite($this->value);
+
+        return $this;
+    }
+
+    /**
+     * Assert that the value is an instance of $value.
+     *
+     * @param mixed $value
+     */
+    public function toBeInstanceOf($value): Expectation
```

Comment:
Should `$value` not be type hinted as a string instead of mixed as that's all that PHPUnit supports for `assertInstanceOf()`? 🤔 It looks like this PHPStan error is related to the Psalm docblocks in PHPUnit.

https://github.com/sebastianbergmann/phpunit/blob/4272d6d7b51159e6de20e3276ef1d508e07ccdc1/src/Framework/Assert.php#L1317-L1339

You should be able to let PHPStan know that this is a `class-string<object>` type with:
```suggestion
     * @param class-string<object> $value
     */
    public function toBeInstanceOf(string $value): Expectation
```